VocaDbModel/Database/Queries/AlbumQueries.cs

```diff
@@ -177,33 +178,34 @@ public AlbumReviewContract AddReview(int albumId, AlbumReviewContract contract)
 				if (contract.Id != 0)
 				{
 					review = ctx.Load<AlbumReview>(contract.Id);
-					if (!review.User.Equals(PermissionContext.LoggedUser))
+					if (!review.Author.Equals(PermissionContext.LoggedUser))
 					{
 						PermissionContext.VerifyPermission(PermissionToken.DeleteComments);
 					}
 				}
 				else
 				{
-					review = ctx.Query<AlbumReview>().FirstOrDefault(r => r.Album.Id == albumId && r.User.Id == PermissionContext.LoggedUserId && r.LanguageCode == contract.LanguageCode);
+					review = ctx.Query<AlbumReview>().FirstOrDefault(r => r.EntryForComment.Id == albumId && r.Author.Id == PermissionContext.LoggedUserId && r.LanguageCode == contract.LanguageCode);
 				}
 
 				// Create
-				if (review == null)
+				if (review == null || review.Deleted)

```

Comment:
What happens if the review is deleted and a new review is posted? Won't this still return the deleted review even after that and then make a new review instead of updating?

1. User posts review (review is created)
2. User deletes review (review is marked deleted)
3. User posts new review (review is created)
4. User updates review (???)

Reply:
Users can successfully update the new review. The `else` block gets called if and only if `review != null && !review.Deleted` (De Morgan's laws).

Reply:
I didn't notice the if block before. So when updating a review, the ID is always non-null and the review gets loaded by ID. Only when creating a new review it's using the query (```ctx.Query<AlbumReview>().FirstOrDefault...```). I was thinking it would use the query every time, which wouldn't work for deleted reviews.

Could you add unit tests for this, since it became more complicated now?

Reply:
It's also ok to have a separate issue for the tests, but I think it should be tested.